homeassistant/components/rest/sensor.py

```diff
@@ -163,32 +162,9 @@ def _update_from_rest_data(self) -> None:
         value = self.rest.data_without_xml()
 
         if self._json_attrs:
-            if value:
-                try:
-                    json_dict = json_loads(value)
-                    if self._json_attrs_path is not None:
-                        json_dict = jsonpath(json_dict, self._json_attrs_path)
-                    # jsonpath will always store the result in json_dict[0]
-                    # so the next line happens to work exactly as needed to
-                    # find the result
-                    if isinstance(json_dict, list):
-                        json_dict = json_dict[0]
-                    if isinstance(json_dict, dict):
-                        attrs = {
-                            k: json_dict[k] for k in self._json_attrs if k in json_dict
-                        }
-                        self._attr_extra_state_attributes = attrs
-                    else:
-                        _LOGGER.warning(
-                            "JSON result was not a dictionary"
-                            " or list with 0th element a dictionary"
-                        )
-                except ValueError:
-                    _LOGGER.warning("REST result could not be parsed as JSON")
-                    _LOGGER.debug("Erroneous JSON: %s", value)
-
-            else:
-                _LOGGER.warning("Empty reply found when expecting JSON data")
+            self._attr_extra_state_attributes = parse_json_attributes(
+                value, self._json_attrs, self._json_attrs_path
+            )
```

Comment:
Note:
- before #97396, `self._attr_extra_state_attributes` was reset to an empty dict if there was an error parsing the new attributes
- after #97396, the previous attributes were kept if there was an error parsing the new attributes

I am not sure that the change in 97396 was intentional, so this PR returns to the previous behavior of clearing the previous attributes on failure.